metrics/ci_integrations/lib/destination/firestore/config/validation_delegate/firestore_destination_validation_delegate.dart

```diff
@@ -80,6 +95,73 @@ class FirestoreDestinationValidationDelegate implements ValidationDelegate {
     return const FieldValidationResult.success();
   }
 
+  /// Validates the given [firebaseProjectId].
+  Future<FieldValidationResult> validateFirebaseProjectId(
+    FirebaseAuthCredentials credentials,
+    String firebaseProjectId,
+  ) async {
+    try {
+      final firestore = await _createFirestore(
+        credentials,
+        firebaseProjectId,
+      );
+
+      await firestore.collection('projects').getDocuments();
```

Comment:
Would it be better to read a concrete document here to avoid redundant document reads since the `projects` collection may contain a lot of documents? Or maybe we can limit this query to return only one document?